File: files/CSEC 04.08.py
import pandas as pd

path = "files/Excel Challenge 4th August.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows = 1, nrows = 6)
test = pd.read_excel(path,  usecols="E:F", skiprows = 1, nrows= 3)
test.columns = test.columns.str.replace('.1', '')

result = input.copy()
result['Customers'] = result['Customers'].str.split('; ')
result = result.explode('Customers')
result['Customers'] = result['Customers'].str.strip()
result['count'] = result.groupby(['Customers', 'Date'])['Customers'].transform('count')
result = result[result['count'] > 1].drop_duplicates()
result = result.groupby('Date')['Customers'].apply(lambda x: '; '.join(x)).reset_index(name='Repeat Customers')

print(result.equals(test)) # True

File: files/CSEC 04.08.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge 4th August.xlsx"
input = read_excel(path, range = "B2:C8")
test  = read_excel(path, range = "E2:F5")

result = input %>%
  separate_rows(Customers, sep = "; ") %>%
  mutate(Customers = str_trim(Customers)) %>%
  mutate(count = n(), .by = c(Customers, Date)) %>%
  filter(count > 1) %>%
  distinct() %>%
  summarise(`Repeat Customers` = str_c(Customers, collapse = "; "), .by = Date)

identical(result, test)
# [1] TRUE